pkg/storage/replica.go

```diff
@@ -618,6 +618,10 @@ func (r *Replica) init(
 func (r *Replica) initRaftMuLockedReplicaMuLocked(
 	desc *roachpb.RangeDescriptor, clock *hlc.Clock, replicaID roachpb.ReplicaID,
 ) error {
+	// Replicas without a replicaID don't have a sideloaded storage. If the
+	// replicaID is nontrivial, we'll recreate the storage later.
+	r.raftMu.sideloaded = nil
```

Comment:
This actually breaks all sorts of things. Have to check what's really going on here, but it seems replicas process Raft commands (log truncation) before they have their sideloaded storage initialized.